### PR TITLE
job_conf related logic refactor part 1

### PR DESCRIPTION
**Pull request by matthewrmshin**
`job_conf` in `TaskProxy`:
* is now a local variable with very limited scope.
* will no longer be populated for poll/kill.

Motivation: while looking at the logic for #1413, it appears that we have been populating `job_conf` for poll/kill for no reason. (It has not been necessary for a while now.) With that gone, the only remaining purpose of `job_conf` is for writing the job file, so it will now only be generated just before writing the job file.

(I think part 2 will be to see if we can remove `job_conf` all together.)

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@hjoliver @benfitzpatrick please review.

**Comment by hjoliver** [comment]
Review 1 - good, no problems found, test battery passes (on my laptop VM, can't currently test with remote hosts etc.)

File: lib/cylc/task_proxy.py [hunk 1]
```diff
@@ -827,7 +808,6 @@ def job_submission_failed(self):
             "time_submit_exit": get_current_time_string(),
             "submit_status": 1,
         })
-        self.submit_method_id = None
```
Comment (benfitzpatrick) [inline lib/cylc/task_proxy.py]: Don't you want to pop `summit_method_id` from self.summary?

**Comment by benfitzpatrick** [comment]
One comment, otherwise fine.

File: lib/cylc/task_proxy.py (hunk 1 above)
Comment (matthewrmshin) [inline lib/cylc/task_proxy.py]: Done.

**Comment by matthewrmshin** [comment]
@benfitzpatrick comment addressed.

**Comment by benfitzpatrick** [comment]
(Can do self.summary.pop('submit_method_id', None))

**Comment by matthewrmshin** [comment]
But `del` should be faster (as we don't need the return value).

**Comment by matthewrmshin** [comment]
@benfitzpatrick is this now good to go?

**Comment by hjoliver** [comment]
(I was just going to ask that myself)

**Comment by benfitzpatrick** [comment]
Yep

**Pull request merged**